assets/js/pages/UpgradablePackagesPage/UpgradablePackages.jsx

```diff
@@ -15,14 +17,67 @@ export default function UpgradablePackages({
   onLoad = noop,
 }) {
   const [search, setSearch] = useState('');
+  const enrichedPackages = upgradablePackages.map((packageDetails) => {
+    const { name, from_version, from_release, to_version, to_release, arch } =
+      packageDetails;
 
-  const displayedPackages = upgradablePackages.filter(
+    return {
+      ...packageDetails,
+      installed_package: `${name}-${from_version}-${from_release}.${arch}`,
+      latest_package: `${name}-${to_version}-${to_release}.${arch}`,
+    };
+  });
+  const displayedPackages = enrichedPackages.filter(
     ({ name, patches }) =>
       containsSubstring(name, search) ||
       patches
         .map(({ advisory }) => containsSubstring(advisory, search))
         .includes(true)
   );
+  const [csvURL, setCsvURL] = useState(null);
+
+  const hasPatches = (packages) =>
+    packages.some((packageDetails) => !isEmpty(packageDetails.patches));
+  const createCsvPackages = (packages) =>
+    packages.map((packageDetails) => {
+      const advisories = packageDetails.patches.map((it) => it.advisory);
+      return {
+        ...packageDetails,
+        patches: advisories,
+      };
+    });
+
+  useEffect(() => {
+    setCsvURL(
+      hasPatches(enrichedPackages)
+        ? URL.createObjectURL?.(
+            new File(
+              [
+                Papa.unparse(
+                  {
+                    fields: ['installed_package', 'latest_package', 'patches'],
+                    data: hasPatches(enrichedPackages)
+                      ? createCsvPackages(enrichedPackages)
+                      : [],
+                  },
+                  { header: true }
+                ),
+              ],
+              `${hostName}-patches.csv`,
+              { type: 'text/csv' }
+            )
+          )
+        : null
+    );
+
+    return () => {
+      if (csvURL) {
+        URL.revokeObjectURL(csvURL);
+      } else {
+        setCsvURL(null);
+      }
+    };
+  }, [patchesLoading]);
```

Comment:
I would change this to `[enrichedPatches]` so the effect is executed everytime this value is changed.